jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererFileCrawlerTest.kt

```diff
@@ -22,31 +22,115 @@ import software.aws.toolkits.jetbrains.services.codewhisperer.util.JavaCodeWhisp
+// TODO: Make different language file crawler different files and move to language/ folder
+class CodeWhispererFileCrawlerTest {
     @JvmField
     @Rule
-    val projectRule: CodeInsightTestFixtureRule = projectRule
+    val projectRule: CodeInsightTestFixtureRule = CodeInsightTestFixtureRule()
+
+    lateinit var sut: CodeWhispererFileCrawler
 
     lateinit var fixture: CodeInsightTestFixture
     lateinit var project: Project
 
-    open fun setup() {
+    @Before
+    fun setup() {
         fixture = projectRule.fixture
         project = projectRule.project
     }
+
+    @Test
+    fun `searchRelevantFileInEditors should exclude target file itself and files with different file extension`() {
+        val targetFile = fixture.addFileToProject("Foo.java", "I have 10 Foo in total, Foo, Foo, Foo, Foo, Foo, Foo, Foo, Foo, Foo")
+
+        val file0 = fixture.addFileToProject("file0.py", "I have 7 Foo, Foo, Foo, Foo, Foo, Foo, Foo, but I am a pyfile")
+        val file1 = fixture.addFileToProject("File1.java", "I have 4 Foo key words : Foo, Foo, Foo")
+        val file2 = fixture.addFileToProject("File2.java", "I have 2 Foo Foo")
+        val file3 = fixture.addFileToProject("File3.java", "I have only 1 Foo")
+        val file4 = fixture.addFileToProject("File4.java", "bar bar bar, i have a lot of bar")
+
+        runInEdtAndWait {
+            fixture.openFileInEditor(targetFile.virtualFile)
+            fixture.openFileInEditor(file0.virtualFile)
+            fixture.openFileInEditor(file1.virtualFile)
+            fixture.openFileInEditor(file2.virtualFile)
+            fixture.openFileInEditor(file3.virtualFile)
+            fixture.openFileInEditor(file4.virtualFile)
+        }
+
+        listOf(
+            JavaCodeWhispererFileCrawler,
+            PythonCodeWhispererFileCrawler,
+            TypescriptCodeWhispererFileCrawler,
+            JavascriptCodeWhispererFileCrawler
+        ).forEach {
+            sut = it
+
+            val result = CodeWhispererFileCrawler.searchRelevantFileInEditors(targetFile) { psiFile ->
+                psiFile.virtualFile.content().split(" ")
+            }
+            assertThat(result).isEqualTo(file1.virtualFile)
+        }
+    }
+
+    @Test
+    fun `searchKeywordsInOpenedFile is language agnostic`() {
+        sut = JavaCodeWhispererFileCrawler
+
+        val targetFile = fixture.addFileToProject("Foo.ts", "I have 10 Foo in total, Foo, Foo, Foo, Foo, Foo, Foo, Foo, Foo, Foo")
+
+        val file0 = fixture.addFileToProject("file0.java", "I have 7 Foo, Foo, Foo, Foo, Foo, Foo, Foo, but I am a pyfile")
+        val file1 = fixture.addFileToProject("File1.ts", "I have 4 Foo key words : Foo, Foo, Foo")
+        val file2 = fixture.addFileToProject("File2.ts", "I have 2 Foo Foo")
+        val file3 = fixture.addFileToProject("File3.ts", "I have only 1 Foo")
+        val file4 = fixture.addFileToProject("File4.ts", "bar bar bar, i have a lot of bar")
```

Comment:
i recommend documenting the expectation that:
```
on Intellij Community, users are still able to trigger CodeWhisperer service on .js .ts files 
```

on this test so we dont forget

Reply:
sg